core/controller/src/main/scala/whisk/core/controller/Controller.scala

```diff
@@ -185,7 +187,49 @@ object Controller {
     }
 
     def main(args: Array[String]): Unit = {
-        implicit val actorSystem = ActorSystem("controller-actor-system")
+
+        // read controller parameters from env variables
+        val port = sys.env("AKKA_CLUSTER_PORT")
```

Comment:
Let's plan for this initialization logic to differ when deploying with Mesos and Kube.